Added .NET example to jobs HowTo page (#4385)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/jobs/howto-schedule-and-handle-triggered-jobs.md

In your code, set up and schedule jobs within your application.

{{< tabs ".NET" "Go" >}}

{{% codetab %}}

<!-- .net -->

The following .NET SDK code sample schedules the job named `prod-db-backup`. Job data is housed in a backup database (`"my-prod-db"`) and is schedueled with `ScheduleJobAsync` from the Dapr Jobs client.  This provides the `jobData` which includes:
- The backup `Task` name
- The backup tasks's `Metadata` including:
  - The database name (`DBName`)
  - The database location (`BackupLocation`)

```csharp
using System.Text;
using System.Text.Json;
using System.Text.Json.Serialization;
using Dapr.Jobs;
using Dapr.Jobs.Extensions;
using Dapr.Jobs.Models;

var builder = WebApplication.CreateBuilder(args);
builder.Services.AddDaprJobsClient();
var app = builder.Build();

// In practice, this would likely be in another service, but for completeness, we'll put this sample in a single file
var DoProdDbBackupHandler = (DbBackupJobPayload? payload) => {
	//Do something
};

> // Set a handler for incoming jobs

Reviewer: Little confused on this comment. This is "configure a map of handlers that are then called" Trying to be clear where the call back from the jobs schedule happens, since that is in `JobsController` yes?

Author: Well, it's the problem with making condensed examples in that I wanted to have a feature-complete demo though in practice, the handler itself would likely be handled in an injected service. I was planning on updating the example to have a more feature-complete sample (and perhaps link to it from here) that shows a more real-world scenario.

Because the Go example in this documentation splits between the SDK, HTTP, and gRPC samples (with a blurb calling out SDKS again at the bottom), I was trying to follow the same format wherein the SDK favors DI-based Jobs client registration and ASP.NET Core mapping (but then have a one-file demo) and then put the HTTP version in the controllers.

var daprJobsClient = scope.ServiceProvider.GetRequiredService<DaprJobsClient>();

//Set up the job payload
var jobPayload = new DbBackupJobPayload("db-backup", new("my-prod-db", "/backup-dir"));
var payloadBytes = JsonSerializer.SerializeToUtf8Bytes(jobPayload);

//Schedule the job
await daprJobsClient.ScheduleJobAsync("prod-db-backup", DaprJobSchedule.FromExpression("@every 1s"), payloadBytes, repeats: 10);

//Specify the records and their JSON property names comprising the sample job payload
sealed record DbBackupJobPayload(
	[property: JsonPropertyName("Task")] string Task, 
	[property: JsonPropertyName("Metadata")] DatabaseMetadata Metadata);
sealed record DatabaseMetadata(
	[property: JsonPropertyName("DBName")] string Name, 
	[property: JsonPropertyName("BackupLocation")] string BackupLocation);
```

The job is using the free-form `@every` expressions to let the developer indicate that the job should be triggered `@every 1s` and sets the optional `repeats` argument to indicate the maximum number of times the job should be triggered and sent back to the app; in this scenario, we set a value of 10 indicating that the job should be triggered a maximum of ten times.

> At the trigger time, the registered handler will be invoked and will execute the desired business logic for the job. In this sample, we apply a switch statement with a value known at compile-time, but in a real-world app, this would very well perform a more dynamic service lookup in order to determine how any given job name should be handled.

Reviewer: ```suggestion
At the trigger time, the registered handler is invoked and executes the desired business logic for the job. In this example, a switch statement is used with a value known at compile-time, but in typical production code it is preferred to have a dynamic service lookup in order to determine how any given job name should be handled.
```

Reviewer: It would be good to explain `DaprJobDetails` and how this is used

Author: Certainly. I think I may rewrite some chunks of this to also provide a better example for `DaprJobSchedule`. In this writing, I was just mirroring the schedule used by the Go demo, but there are some helper methods in this class that should help developers properly format these expressions without necessarily knowing the various expression syntaxes.

I also mildly dispute the last suggestion as this is code simply presented as a simple proof-of-concept. Ideally, production code mirrors development code, so I would assert it is a better fit to refer to this as real-world vs demonstration code as opposed to the suggested "production code".

> The `MapDaprScheduledJobHandler` accepts a delegate that works much like the ASP.NET Core minimal API parameter binding logic. The first parameter will always contain the name of the job, the second will reflect the job details provided in the callback and the last parameter must be a cancellation token. Otherwise, the developer is free to provide any additional services registered via dependency injection before the cancellation token argument and they will be injected at runtime.

Reviewer: ```suggestion
The `MapDaprScheduledJobHandler` accepts a delegate that works much like the ASP.NET Core minimal API parameter binding logic. The first parameter is the name of the job, the second parameter is job details provided in the callback and the last parameter is a cancellation token. You are free to provide any additional services registered via dependency injection before the cancellation token argument and they are injected at runtime.
```

Author: I'm fine with this except "the second parameter is job details" doesn't flow right to me. Would you be more accepting of "the second parameter contains job details" with a description in a prior paragraph of what those job details might look like?

#### HTTP

> When you create a job using Dapr's Jobs API, Dapr will automatically assume there is an endpoint available at `/job/<job-name>`. For instance, if you schedule a job named `test`, Dapr expects you application to listen for job events at `job/test`. Ensure your application has a handler set up for this endpoint to process the job when it is triggered. For example:

Reviewer: ```suggestion
When you create a job using Jobs API, Dapr assumes there is an endpoint available at `/job/<job-name>`. For example, if you schedule a job named `test`, Dapr expects your application to listen for job events at `job/test`. Ensure your application has a handler set up for this endpoint to process the job when it is triggered. For example:
```
